test/blackbox/openid_claim_mapping.bats

```diff
@@ -0,0 +1,430 @@
+# Note: Intended to be run as "make run-blackbox-tests" or "make run-blackbox-ci"
+#       Makefile target installs & checks all necessary tooling
+#       Extra tools that are not covered in Makefile target needs to be added in verify_prerequisites()
+
+load helpers_zot
+load helpers_wait
+load ../port_helper
+
+function verify_prerequisites {
+    if [ ! $(command -v curl) ]; then
+        echo "you need to install curl as a prerequisite to running the tests" >&3
+        return 1
+    fi
+
+    if [ ! $(command -v jq) ]; then
+        echo "you need to install jq as a prerequisite to running the tests" >&3
+        return 1
+    fi
+
+    if [ ! $(command -v docker) ]; then
+        echo "you need to install docker as a prerequisite to running the tests" >&3
+        return 1
+    fi
+
+    return 0
+}
+
+DEX_CONTAINER_NAME="dex_oidc_claim_mapping_test"
+
+function setup_file() {
+    # Verify prerequisites are available
+    if ! $(verify_prerequisites); then
+        exit 1
+    fi
+
+    # Pre-allocate all ports we'll need
+    dex_port=$(get_free_port_for_service "dex")
+    zot_port_1=$(get_free_port_for_service "zot_preferred_username")
+    zot_port_2=$(get_free_port_for_service "zot_email")
+    zot_port_3=$(get_free_port_for_service "zot_sub")
+    zot_port_4=$(get_free_port_for_service "zot_default")
+    
+    echo ${dex_port} > ${BATS_FILE_TMPDIR}/dex.port
+
+    # Setup dex OIDC provider
+    local dex_config_file=${BATS_FILE_TMPDIR}/dex_config.yaml
+
+    # Create dex config with mockCallback connector and all redirect URIs
+    cat > ${dex_config_file}<<EOF
+issuer: http://127.0.0.1:${dex_port}/dex
+
+storage:
+  type: memory
+
+web:
+  http: 0.0.0.0:${dex_port}
+
+staticClients:
+  - id: zot-client
+    redirectURIs:
+      - 'http://127.0.0.1:${zot_port_1}/zot/auth/callback/oidc'
+      - 'http://127.0.0.1:${zot_port_2}/zot/auth/callback/oidc'
+      - 'http://127.0.0.1:${zot_port_3}/zot/auth/callback/oidc'
+      - 'http://127.0.0.1:${zot_port_4}/zot/auth/callback/oidc'
+    name: 'zot'
+    secret: ZXhhbXBsZS1hcHAtc2VjcmV0
+
+connectors:
+  - type: mockCallback
+    id: mock
+    name: Example
```

Comment:
You probably want to add preferredUsername: "testuser" etc under here to actually test